### PR TITLE
#10212: Enabling CPP post commit tests on BH

### PR DESCRIPTION
**Pull request by abhullar-tt**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/10212

### What's changed
Enabling post commit for BH

### Checklist
- [x] [Post commit CI passes](https://github.com/tenstorrent/tt-metal/actions/runs/10288668195)
- [ ] [BH post commit CI](https://github.com/tenstorrent/tt-metal/actions/runs/10292291975) (not passing - debugging machines)

### CODE REVIEW

File: .github/workflows/build-and-unit-tests.yaml [hunk 1]
```diff
@@ -18,6 +18,7 @@ jobs:
           {arch: wormhole_b0, runs-on: ["wormhole_b0", "multi-chip-num-pcie-1", "multi-chip-num-chips-1"], name: N150},
           # N300
           {arch: wormhole_b0, runs-on: ["wormhole_b0", "multi-chip-num-pcie-1", "multi-chip-num-chips-2"], name: N300},
+          {arch: blackhole, runs-on: ["blackhole"], name: BH},
```
Comment (ttmchiou) [inline .github/workflows/build-and-unit-tests.yaml:21]: can be removed, 
Will make relevant changes for CI in this PR
https://github.com/tenstorrent/tt-metal/pull/10888

Comment (abhullar-tt) [reply .github/workflows/build-and-unit-tests.yaml]: removed

**Review by ttmchiou — APPROVED** [review]

File: tt_metal/impl/dispatch/kernels/cq_dispatch.cpp [hunk 2]
```diff
@@ -162,7 +162,7 @@ void process_write_host_h() {
     // We will send the cmd back in the first X bytes, this makes the logic of reserving/pushing completion queue
     // pages much simpler since we are always sending writing full pages (except for last page)
     uint32_t length = cmd->write_linear_host.length;
-    DPRINT << "process_write_host_h: " << length << ENDL();
+    // DPRINT << "process_write_host_h: " << length << ENDL();
```
Comment (tt-dma) [inline tt_metal/impl/dispatch/kernels/cq_dispatch.cpp:165]: Any reason why we're commenting out the dprints on cq_dispatch specifically?

Comment (abhullar-tt) [reply tt_metal/impl/dispatch/kernels/cq_dispatch.cpp:165]: I was running into code size issues when running watcher/dprint fixtures

**Review by aliuTT — APPROVED** [review]

**Review by uaydonat — APPROVED** [review]

**Review by tt-rkim — APPROVED** [review]

**Comment by tt-rkim** [comment]
We should probably make necessary updates to the `infra/data_collection` scripts and add `blackhole` as an option.

**Comment by abhullar-tt** [comment]
> We should probably make necessary updates to the `infra/data_collection` scripts and add `blackhole` as an option.

Opened up https://github.com/tenstorrent/tt-metal/issues/11238 for this

**Pull request merged**